packages/client-utils/src/signing-smart-wallet-kit.ts

```diff
@@ -44,21 +62,25 @@ export const makeSigningSmartWalletKit = async (
   // Omit deprecated utilities
   const { storedWalletState: _, ...swk } = walletUtils;
 
+  const getLastUpdate = freeze(() => swk.getLastUpdate(address));
   const query = {
     readPublished: swk.readPublished,
     vstorage: swk.vstorage,
-    getLastUpdate: () => swk.getLastUpdate(address),
-    getCurrentWalletRecord: () => swk.getCurrentWalletRecord(address),
-    pollOffer: (
-      ...args: Parameters<SmartWalletKit['pollOffer']> extends [
-        any,
-        ...infer Rest,
-      ]
-        ? Rest
-        : never
-    ) => swk.pollOffer(address, ...args),
+    getLastUpdate,
+    getCurrentWalletRecord: freeze(() => swk.getCurrentWalletRecord(address)),
```

Comment:
this approach of freezing each inline clutters the code and could be forgotten.

I suppose you're avoiding `harden` in case it's not available but client-utils already depends on harden. Could you just `harden(query)`?

Reply:
I had assumed the existing use of `freeze` rather than `harden` was for precisely that reason, but I see now that it was just naïvely accepting https://github.com/Agoric/agoric-sdk/pull/11778#discussion_r2288917024 . The code does not in fact load without lockdown, so switching everything to `harden` makes sense.